Optimal_bounding_box/include/CGAL/Optimal_bounding_box/obb.h

```diff
@@ -0,0 +1,253 @@
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0+
+//
+//
+// Author(s)     : Konstantinos Katrioplas
+
+#ifndef CGAL_OPTIMAL_BOUNDING_BOX_OBB_H
+#define CGAL_OPTIMAL_BOUNDING_BOX_OBB_H
+
+#include <CGAL/Optimal_bounding_box/population.h>
+#include <CGAL/Optimal_bounding_box/evolution.h>
+#include <CGAL/Optimal_bounding_box/helper.h>
+#include <vector>
+#include <CGAL/boost/graph/helpers.h>
+#include <CGAL/Bbox_3.h>
+#include <CGAL/Iso_cuboid_3.h>
+#include <CGAL/convex_hull_3.h>
+#include <CGAL/Polyhedron_3.h>
+#include <CGAL/Simple_cartesian.h>
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#ifdef OBB_BENCHMARKS
+#include <CGAL/Timer.h>
+#endif
+
+#if defined(CGAL_EIGEN3_ENABLED)
+#include <CGAL/Eigen_linear_algebra_traits.h>
+#endif
+
+namespace CGAL {
+namespace Optimal_bounding_box {
+
+
+#if defined(CGAL_EIGEN3_ENABLED)
+typedef CGAL::Eigen_linear_algebra_traits Linear_algebra_traits;
+#endif
+
+
+// works on matrices only
+/// \cond SKIP_IN_MANUAL
+template <typename Vertex, typename Matrix>
+void post_processing(const Matrix& points, Vertex& R, Matrix& obb)
+{
+  CGAL_assertion(points.cols() == 3);
+  CGAL_assertion(R.rows() == 3);
+  CGAL_assertion(R.cols() == 3);
+  CGAL_assertion(obb.rows() == 8);
+  CGAL_assertion(obb.cols() == 3);
+
+  // 1) rotate points with R
+  Matrix rotated_points(points.rows(), points.cols());
+  rotated_points = points * Linear_algebra_traits::transpose(R);
+
+  // 2) get AABB from rotated points
+  typedef CGAL::Simple_cartesian<double> K;
+  typedef K::Point_3 Point;
+  std::vector<Point> v_points; // Simplex -> std::vector
+  for(std::size_t i = 0; i < rotated_points.rows(); ++i)
+  {
+    Point p(rotated_points(i, 0), rotated_points(i, 1), rotated_points(i, 2));
+    v_points.push_back(p);
+  }
+  CGAL::Bbox_3 bbox;
+  bbox = bbox_3(v_points.begin(), v_points.end());
+  K::Iso_cuboid_3 ic(bbox);
+
+  // Matrix is [dynamic, 3] at compile time, so it needs allocation of rows at run time.
+  Matrix aabb(8, 3);
+
+  for(std::size_t i = 0; i < 8; ++i)
+  {
+    aabb.set_coef(i, 0, ic[i].x());
+    aabb.set_coef(i, 1, ic[i].y());
+    aabb.set_coef(i, 2, ic[i].z());
+  }
+
+  // 3) apply inverse rotation to rotated AABB
+  obb = aabb * R;
+}
+/// \endcond
+
+/// \ingroup OBB_grp
+/// calculates the optimal bounding box.
+///
+/// @tparam Point the point type
+/// @tparam LinearAlgebraTraits a model of `LinearAlgebraTraits`
+///
+/// @param points the input points that are included in the optimal bounding box.
+/// @param obb_points the eight points of the optimal bounding box to be calculated.
+/// @param use_ch a bool flag to indicating whether to use the convex hull of the input points
+/// as an optimization step.
+template <typename Point, typename LinearAlgebraTraits>
+void find_obb(std::vector<Point>& points,
```

Comment:
When we had discussed the API, we wanted a function that takes a range of points  and returns the transformation matrix. @sloriot did you have a `CGAL::Affine_transformation_3` in mind or a `LinearAlgebraTraits::Matrix3d` ?